compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -7025,6 +7037,30 @@ private BType checkInvocationArgs(BLangInvocation iExpr, List<BType> paramTypes,
         }
     }
 
+    private void populateIncludedRecordParams(BVarSymbol param, HashSet<String> includedRecordFields,
+                                              List<String> includedRecordParamNames) {
+        Set<String> fields = new HashSet<>(((BRecordType) Types.getReferredType(param.type)).fields.keySet());
+        fields.removeIf(field -> !includedRecordParamNames.contains(field));
+        includedRecordFields.addAll(fields);
```

Comment:
Why aren't we just iterating over `((BRecordType) Types.getReferredType(param.type)).fields.keySet()` and directly adding to `includedRecordFields`? We don't need the extra set?